streams/base64_decoder_stream.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+// Based on https://github.com/mazira/base64-stream
+
+import { decodeBase64 } from "@std/encoding/base64";
+
+/**
+ * A transform stream that decodes a stream of base64-encoded strings.
+ *
+ * @example
+ * ```ts
+ * import { Base64DecoderStream } from "@std/streams/base64-decoder-stream";
+ *
+ * const stream = ReadableStream.from([
+ *   "aGVsbG8gd29ybGQ=",
+ *   "Zm9vYmFy",
+ * ]);
+ *
+ * const decodedStream = stream.pipeThrough(new Base64DecoderStream());
+ * ```
+ */
+export class Base64DecoderStream extends TransformStream<string, Uint8Array> {
+  /** The remain data from the previous chunk. */
+  #remain: string = "";
+
+  constructor() {
+    super({
+      transform: (chunk, controller) => {
+        chunk = this.#remain + chunk.replaceAll(/(\r\n|\r|\n)/g, "");
```

Comment:
I'm not sure why we need to replace line-breaks here

Reply:
There will be some cases where the encoded data is split into lines, for example, when working with MIME

Reply:
hmm maybe it's useful in that case, but it could be unnecessary perf penalty for the case when there's no line-breaks in input.. I wonder if it's worth making this replacement optional.

Reply:
What type of MIME type would insert line breaks in what is meant to be one continuous base64 string?

Reply:
The MIME mentioned here is a standard format of email messages. According to [RFC 2045](https://datatracker.ietf.org/doc/html/rfc2045#section-6.8):
> The encoded output stream must be represented in lines of no more than 76 characters each. All line breaks or other characters not found in Table 1 must be ignored by decoding software.

Reply:
That requirement of 76 characters per line isn't within the Base64 encoding spec https://datatracker.ietf.org/doc/html/rfc2045#section-6.7 so arguable such behaviour should be excluded from it and handled either before or after when necessary.